pilot/pkg/config/kube/crd/controller/controller.go

```diff
@@ -348,10 +348,10 @@ func (c *controller) List(typ, namespace string) ([]model.Config, error) {
 			continue
 		}
 
-		config, err := crd.ConvertObject(s, item, c.client.domainSuffix)
+		config, err := crd.StrictConvertObject(s, item, c.client.domainSuffix)
```

Comment:
I think it is safe to have this validation by default - whether galley is used or not. But if folks, think it is not needed always, I can add a feature flag to enable/disable this.

Reply:
Do we want to validate at read time (in List) rather than when the CRD is being written? So we don't even store the CRD internally if its invalid

Reply:
hmm, I think we are already doing it here https://github.com/istio/istio/blob/master/pilot/pkg/config/kube/crd/controller/controller.go#L154

Not sure why List is returning invalid one. Here is the error we get 
```
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0xe3b5f9]
goroutine 758 [running]:
istio.io/istio/pilot/pkg/model.(*PushContext).initVirtualServices(0xc0013e8480, 0xc00096e400, 0x0, 0x0)
    istio.io/istio@/pilot/pkg/model/push_context.go:1044 +0x599
istio.io/istio/pilot/pkg/model.(*PushContext).createNewContext(0xc0013e8480, 0xc00096e400, 0x0, 0x0)
    istio.io/istio@/pilot/pkg/model/push_context.go:736 +0x64
istio.io/istio/pilot/pkg/model.(*PushContext).InitContext(0xc0013e8480, 0xc00096e400, 0xc00055a120, 0xc0017bca80, 0x0, 0x0)
    istio.io/istio@/pilot/pkg/model/push_context.go:718 +0x150
istio.io/istio/pilot/pkg/proxy/envoy/v2.(*DiscoveryServer).Push(0xc0008c9180, 0xc0017bca80)
```

Reply:
Do we still need validation in add and update funcs here https://github.com/istio/istio/blob/master/pilot/pkg/config/kube/crd/controller/controller.go#L154. - My reading of k8s code, it seems after list, it invokes add and update listeners - so I think it is redundant in add/update listeners. What do you think?

Reply:
I went ahead and removed from add/update funcs, LMK if that is an issue. I can put it back.